File: pdb_corner/subset_pdb_by_chain.py
# python3
import argparse
from Bio.PDB import *
# input parameters
ap = argparse.ArgumentParser()
ap.add_argument("-in", "--input_pdb", required=True, help="input pdb file")
ap.add_argument("-model", "--pdb_model", required=False, default= 0, help="model from pdb file to select(integer, default=0)")
ap.add_argument("-chain", "--pdb_chain", required=True, help="chain from pdb file to select")
ap.add_argument("-out", "--output_pdb", required=True, help="output pdb file")
args = vars(ap.parse_args())
# main
parser = PDBParser()
s = parser.get_structure("name", args['input_pdb'])
fill = s[int(args['pdb_model'])][args['pdb_chain']]
io = PDBIO()
io.set_structure(fill)
io.save(args['output_pdb'])

File: pdb_corner/subset_pdb_by_chain.jl
# julia
using ArgParse
using BioStructures
# input parameters function
function parse_commandline()

    s = ArgParseSettings(description = "input a pdb ids from PDB and subset by selecting the model and chain")
    @add_arg_table s begin
        "--id"
            help = "input pdb ids"
        "--model"
            help = "model to make the pdb file for"
            arg_type = Int
            default = 1
            required = false
        "--chain"
            help = "chain in model to make the pdb file for"    
        "--out"
            help = "output pdb file"
    end
    return parse_args(s)
end

function main()
    parsed_args = parse_commandline()
    println(parsed_args)
# main
    downloadpdb(parsed_args["id"]) do fp
        s = read(fp, PDB)
        subpdb = s[parsed_args["model"]][parsed_args["chain"]]
        writepdb(parsed_args["out"], subpdb)
    end
end

main()